specifications/Dutch Auction.md

```diff
@@ -0,0 +1,172 @@
+* Determines the current auction id by increasing a counter.
+* Defines the auction's start block as the current block number.
+* Defines initial and final price
+* Upon execution emit an AuctionStarted event with auction id, initial price and final price
+This operation can only be executed by factory
+
+##### Get Bid Price
+This function will be called when a bidder wants to know the current bid price, and performs the following actions: 
+* Calculates the number of blocks between current block and auction's start block (time elapsed since start).
+* Determines current auction price using the following function:
+```solidity
+    function _dutchAuctionPrice(uint256 blocks) internal view returns (uint256 result) {
+        uint256 _alfa = _startPrice - _finalPrice;
+        uint256 t1 = _alfa * _scaleParameter;
+        uint256 t2 = _decay * blocks + _scaleParameter ** 2;
+        uint256 ratio = t1 / t2;
+        return _finalPrice + ratio;
+    }
+```
+This operation is public
+
+##### Bid for Current Price
+This function will be called when a bidder wants to buy for the bidding price, and performs the following actions: 
+* Emits an event with address of the winner (sender) and auction details
+This operation is public
+
+#### Testing
+This is the expected initial bidding price (in ETH):
+"1000000.0000000000000000000"
+
+These are the expected price values for the first 5 blocks of the started auction (in ETH):
+1. "10000.009504000009504000"
+2. "9984.035063258795446645"
+3. "9968.111577671460859968"
+4. "9952.238803821665555414"
+5. "9936.416499841026992686"
+
+These are the expected price values for the first 30 days of the started auction (in ETH):
+1. "10000.009504000009504000"
+2. "978.866285982781712764"
+3. "514.624662988893909592"
+4. "349.074103769906273554"
+5. "264.112703317144606972"
+6. "212.414015972821832455"
+7. "177.642112150073545999"
+8. "152.653388985233679619"
+9. "133.828247682270598608"
+10. "119.136635928723979872"
+11. "107.351805925299422928"
+12. "97.688742611559180995"
+13. "89.621757717408695849"
+14. "82.785574306346927859"
+15. "76.918477622602351368"
+16. "71.828042286708823671"
+17. "67.369625219603283677"
+18. "63.432401156841503597"
+19. "59.930025099477506044"
+20. "56.794226720583805605"
+21. "53.970316080303145780"
+22. "51.413966821574759682"
+23. "49.088872370342160681"
+24. "46.965010690817608264"
+25. "45.017340899444302336"
+26. "43.224811339681163975"
+27. "41.569595611274020919"
+28. "40.036497691258118620"
+29. "38.612484036944839430"
+30. "37.286312134325050093"
+
+The contract should pass the following tests:
+✓ Should obtain correct bid price at first auction block
+✓ Should obtain correct bid prices through five blocks according to dutch auction curve
+✓ Should restart the auction
+✓ Should get correct prices for the first 30 days
+
+#### Presentation
+
+N/A
+
+#### Security / Risks
+
+Since in this Contract public users do not modify state, no critical risks are detected
```

Comment:
I think there's still some risk around vulnerabilities causing unexpected auction behavior.

- Being able to start an auction via manipulation of factory state
- Being able to get into validator pool for unexpected price, messing with economics
- Validators being put into a bad state with respect to generating the ETHDKG ceremony as a race condition
- Etc.

Reply:
It may need to be the case that if we are adding a validator via a Dutch auction that we stop mining blocks. We will need to be careful about the specific details to ensure no race condition.

Reply:
Added